include/sw/universal/number/unum2/_unum2_impl.hpp

```diff
@@ -0,0 +1,320 @@
+
+// template class reprfsizesizeenting a value in scientific notation, using a template size for the number of fraction bits
+template<unsigned esizesize, unsigned fsizesize, typename bt = uint8_t>
+class unum2 {
+public:
+	static constexpr unsigned UTAGSIZE   = 1 + esizesize + fsizesize;
+	static constexpr unsigned UTAGMASK   = unsigned(~(int64_t(-1) << UTAGSIZE));
+	static constexpr unsigned EBITSMASK  = 1;
+	static constexpr unsigned FBITSMASK  = 2;
+
+	unum2() {}
+
+	// specific value constructor
+	constexpr unum2(const SpecificValue code) {
+		switch (code) {
+		case SpecificValue::maxpos:
+			maxpos();
+			break;
+		case SpecificValue::minpos:
+			minpos();
+			break;
+		default:
+			zero();
+			break;
+		case SpecificValue::minneg:
+			minneg();
+			break;
+		case SpecificValue::maxneg:
+			maxneg();
+			break;
+		}
+	}
+
+	unum2(signed char initial_value)        { *this = initial_value; }
+	unum2(short initial_value)              { *this = initial_value; }
+	unum2(int initial_value)                { *this = initial_value; }
+	unum2(long long initial_value)          { *this = initial_value; }
+	unum2(unsigned long long initial_value) { *this = initial_value; }
+	unum2(float initial_value)              { *this = initial_value; }
+	unum2(double initial_value)             { *this = initial_value; }
+	unum2(long double initial_value)        { *this = initial_value; }
+	unum2(const unum2& rhs)                  { *this = rhs; }
+
+	// assignment operators
+	unum2& operator=(signed char rhs) {
+		return *this = (long long)(rhs);
+	}
+	unum2& operator=(short rhs) {
+		return *this = (long long)(rhs);
+	}
+	unum2& operator=(int rhs) {
+		return *this = (long long)(rhs);
+	}
+	unum2& operator=(long long rhs) {
+		return *this;
+	}
+	unum2& operator=(unsigned long long rhs) {
+		return *this;
+	}
+	unum2& operator=(float rhs) {
+
+		return *this;
+	}
+	unum2& operator=(double rhs) {
+
+		return *this;
+	}
+	unum2& operator=(long double rhs) {
+
+		return *this;
+	}
+
+	// arithmetic operators
+	// prefix operator
+	unum2 operator-() const {				
+		return *this;
+	}
+
+	unum2& operator+=(const unum2& rhs) {
+		return *this;
+	}
+	unum2& operator+=(double rhs) {
+		return *this += unum2(rhs);
+	}
+	unum2& operator-=(const unum2& rhs) {
+
+		return *this;
+	}
+	unum2& operator-=(double rhs) {
+		return *this -= unum2<esizesize, fsizesize>(rhs);
+	}
+	unum2& operator*=(const unum2& rhs) {
+
+		return *this;
+	}
+	unum2& operator*=(double rhs) {
+		return *this *= unum2<esizesize, fsizesize>(rhs);
+	}
+	unum2& operator/=(const unum2& rhs) {
+
+		return *this;
+	}
+	unum2& operator/=(double rhs) {
+		return *this /= unum2<esizesize, fsizesize>(rhs);
+	}
+	unum2& operator++() {
+		return *this;
+	}
+	unum2 operator++(int) {
+		unum2 tmp(*this);
+		operator++();
+		return tmp;
+	}
+	unum2& operator--() {
+		return *this;
+	}
+	unum2 operator--(int) {
+		unum2 tmp(*this);
+		operator--();
+		return tmp;
+	}
+
+	// modifiers
+
+	/// <summary>
```

Comment:
remove the commentary for simple modifiers and selectors. These are header files and thus all the characters in the file are read during compilation. No need to document standard APIs that have a well defined meaning.